### ISSUE TITLE
Added 'autotune' command for automated tuning of alchemical protocol

### ISSUE THREAD

**Issue by jchodera** [choderalab/yank#119]
We should add an `autotune` command to the cli that would attempt to optimize the alchemical protocol to minimize the discrete analog to thermodynamic length by reweighting existing data in the repex stores.

**Comment by Lnaden** [comment]
Post 1.0 issue

**Comment by andrrizzi** [comment]
Resurrecting this thread as we'll have to add this for YANK 1.0 paper.

As I see it, we have two options (not necessarily exclusive) to expose this feature

1. A new command, something like `yank autotune protocol --yaml=yank.yaml` or `yank optimize protocol --yaml=yank.yaml`.
2. Add this directly to the YAML syntax with something like
```yaml
protocols:
  absolute-binding:
    complex:
      alchemical_path: auto
    solvent:
      alchemical_path: auto
```

In both cases, I think the output should be multiple YAML scripts, one for each experiment, rather than a single monolithic, unreadable new script, especially for big combinatorial experiments.

For this to work, I would have to change a little the behavior of `ExperimentBuilder` to make it perform a recursive search into the experiment folders looking for YAML scripts to run. Currently, the single-experiment scripts are saved only for backup purposes. This structure could also simplify parallelizing multiple experiments in the future.

**Comment by jchodera** [comment]
I think there should be two distinct features:
* (1) Construct an initial guess for the alchemical protocol (implemented for yank 1.0)
* (2) Update the alchemical protocol based on an existing simulation (implemented sometime in the future)

For (1), I like the idea of adding the `auto` keyword directly to the protocol description in the YAML. Is it possible that, as YANK *generates* the explicitly substituted YAML file for the experiment it is about to run, it could detect whether `auto` is used and carry out the fast lambda scan to produce an initial protocol guess at that point, and then fill in the explicit protocol in the experiment directory? Or would this make early termination unrecoverable?

**Comment by andrrizzi** [comment]
I agree about the two distinct features.

For (1), I was thinking to implement it exactly that way, possibly in parallel using the `mpi` library. In other words, `ExperimentBuilder` would do sequentially
1. Run the setup pipeline on all molecules/systems that needed it (possibly in parallel in the future).
2. Create a single-experiment YAML script for each combination and save it in its experiment folder. In this phase, YANK would run the automatic alchemical path determination for each experiment that specifies the `auto` keyword (in parallel).
3. Run all the experiments using the single-experiment YAML scripts generated in step 2).

**Comment by jchodera** [comment]
This sounds perfect, @andrrizzi!

**Comment by Lnaden** [comment]
Fixed in #726
